packages/rest/src/rest.server.ts

```diff
@@ -151,7 +151,23 @@ export class RestServer extends Context implements Server, HttpServerLike {
     return this._httpServer ? this._httpServer.listening : false;
   }
 
+  /**
+   * The base url for the server, including the basePath if set. For example,
+   * the value will be 'http://localhost:3000/api' if `basePath` is set to
+   * '/api'.
+   */
   get url(): string | undefined {
+    let serverUrl = this.rootUrl;
+    if (!serverUrl) return serverUrl;
+    serverUrl = serverUrl + (this._basePath || '');
+    return serverUrl;
+  }
+
+  /**
+   * The root url for the server without the basePath. For example, the value
+   * will be 'http://localhost:3000' regardless of the `basePath`.
+   */
+  get rootUrl(): string | undefined {
```

Comment:
Probably `serverUrl` is more descriptive of what it actually is? `rootUrl` sounds more like the functionality provided by `url` above.

Reply:
It's a property on `RestServer`. `RestServer.serverUrl` seems to be confusing and it might be misunderstood as OpenAPI server url/

Reply:
Naming is important and often difficult, thank you @hacksparrow for starting this discussion.

I agree with Raymond's arguments here.

Reply:
@hacksparrow The `rootUrl` is only used to create the request test client at the moment. Let's go with it for now.